libexec/rtld-elf/aarch64/reloc.c

```diff
@@ -167,6 +177,224 @@ _rtld_relocate_nonplt_self(Elf_Dyn *dynp, Elf_Auxinfo *aux)
 }
 #endif /* __CHERI_PURE_CAPABILITY__ */
 
+#if defined(__CHERI_PURE_CAPABILITY__) && defined(RTLD_SANDBOX)
+
+void _rtld_thread_start(struct pthread *);
+void _rtld_sighandler(int, siginfo_t *, void *);
+
+struct tramp {
+	const void *target;
+	uint32_t compart_id;
+	char padding;
```

Comment:
So that the pointer to the `code` member is offset by 1, which in Morello means capability mode code.

This is a dirty hack, but it's the simplest solution I can think of.

Reply:
Yeah no. The simplest, and correct, solution is to add 1 to it when you want to turn it from a data pointer to a code pointer, at the same time as you go and turn it into a sentry. You can even use the MI cheri_capmode helper function (was a no-op on CHERI-MIPS, does a CSetFlags on CHERI-RISC-V, does `| 1` on Morello) to hide that Morello implementation detail and make it self-documenting.